### PR TITLE
nmartinez challenge

### PR DESCRIPTION
**Pull request by NZenitram**

### CODE REVIEW

**Comment by jcasimir** [comment]
👍

File: test/string_validator_test.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+require './lib/string_validator'
+require 'minitest/autorun'
+require 'minitest/pride'
+
+class StringValidatorTest < Minitest::Test
+
+  def test_it_equals_four
+    v = StringValidator.new
+    assert_equal 4, v.num
```
Comment (mikedao) [inline test/string_validator_test.rb:9]: Why should it equal four?

File: lib/string_validator.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+  def num
+    4
+  end
+
+  def validate(string)
+    input = string.split("")
+    comp = input.any? {|x| ["(", "[", "{", ")", "]", "}"].include?(x)}
+    if comp = true
+      input[-1].include?("(") || input[-1].include?("{") || input[-1].include?("[")
```
Comment (mikedao) [inline lib/string_validator.rb:13]: Lets pull out this complex conditional into a predicate method.

File: test/string_validator_test.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    string = "([{}[]])"
+    v = StringValidator.new
+    assert_equal true, v.validate(string)
+  end
+
+  def test_validates_second_string
+    string = "()["
+    v = StringValidator.new
+    assert_equal false, v.validate(string)
+  end
```
Comment (mikedao) [inline test/string_validator_test.rb:28]: Would have loved to see some sad path and edge cases here.

**Review by mikedao — APPROVED** [review]
